src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,465 @@
+                ISyntaxFacts syntaxFacts, AbstractInlineMethodRefactoringProvider inlineMethodRefactoringProvider, SyntaxNode syntaxNode)
+            {
+                for (var node = syntaxNode; node != null; node = node!.Parent)
+                {
+                    // Is there anything missed here?
+                    if (node != null && (
+                        syntaxFacts.IsLocalDeclarationStatement(node)
+                        || inlineMethodRefactoringProvider.IsEmbeddedStatementOwner(syntaxNode)
+                        || syntaxFacts.IsExpressionStatement(node)))
+                    {
+                        return node;
+                    }
+                }
+
+                return syntaxNode;
+            }
+
+            /// <summary>
+            /// Generate a new identifier name. If <param name="identifierName"/> has a number suffix,
+            /// increase it by 1. Otherwise, append 1 to it.
+            /// </summary>
+            private static string GenerateNewName(string identifierName)
+            {
+                var stack = new Stack<char>();
+                for (var i = identifierName.Length - 1; i >= 0; i--)
+                {
+                    var currentCharacter = identifierName[i];
+                    if (char.IsNumber(currentCharacter))
+                    {
+                        stack.Push(currentCharacter);
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+
+                var suffixNumber = stack.IsEmpty() ? 1 : int.Parse(new string(stack.ToArray())) + 1;
+                return identifierName.Substring(0, identifierName.Length - stack.Count) + suffixNumber;
+            }
+        }
+
+        private class VariableDeclaratorOperationVisitor : OperationWalker
+        {
+            private readonly CancellationToken _cancellationToken;
+            private readonly HashSet<ILocalSymbol> _localSymbols;
+
+            public VariableDeclaratorOperationVisitor(CancellationToken cancellationToken)
+            {
+                _cancellationToken = cancellationToken;
+                _localSymbols = new HashSet<ILocalSymbol>();
+            }
+
+            public ImmutableArray<ILocalSymbol> FindAllLocalSymbols(IOperation? operation)
+            {
+                if (operation != null)
+                {
+                    Visit(operation);
+                }
+
+                return _localSymbols.ToImmutableArray();
+            }
+
+            public override void Visit(IOperation operation)
+            {
+                _cancellationToken.ThrowIfCancellationRequested();
+                if (operation is IVariableDeclaratorOperation variableDeclaratorOperation)
+                {
+                    _localSymbols.Add(variableDeclaratorOperation.Symbol);
+                }
+
+                base.Visit(operation);
+            }
```

Comment:
i skipped a lot of this as well.  coudl use a nice story walking through things here.

Reply:
😃Sure